src/editor/components/components/StreetSegmentSidebar.js

```diff
@@ -0,0 +1,90 @@
+import PropTypes from 'prop-types';
+import PropertyRow from './PropertyRow';
+
+const StreetSegmentSidebar = ({ entity }) => {
```

Comment:
i feel like there's a way to create a `managedsidebar` that takes entity, component name, and the properties that are meant to be rendered as some prop... That would DRY up the logic for this and `ManagedStreetSidebar` and clean up the `Sidebar` code as well

Reply:
i dont think you need to do this though. it would be interesting to pass this into cursor and see if they can clean it up

Reply:
Indeed, instead of `ManagedStreetSidebar` and `StreetSegmentSidebar`, create a `ComponentSidebar` component with props
- entity
- componentName
- uiSchema: [{property: "length", label: "Street Length"}, {property: "width", label: "Street Width"}] for ManagedStreetSidebar for example

Reply:
yes for later